app/controllers/discussions_controller.rb

```diff
@@ -77,9 +78,19 @@ def unfavorite
     redirect_to discussions_path(discussion_id: params[:discussion_id], search: params[:search])
   end
 
+  def upvote
```

Comment:
Make sure to handle the edge case when a user tries to upvote one of their own responses/discussions they created. Got this error: ArgumentError - comparison of Array with Array failed